### PR TITLE
mpv: 0.36.0 -> 0.37.0 

### LINKED ISSUE #270002: Update request: mpv 0.36.0 → 0.37.0
https://github.com/mpv-player/mpv/releases/tag/v0.37.0
- Package name: mpv
- Latest released version: 0.37.0

- Current version on the unstable channel: 0.36.0
- Current version on the stable/release channel: 0.35.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@AndersonTorres @fpletz @globin @Ma27 @tadeokondrak 

-----

Note for maintainers: Please tag this issue in your PR.

**Priorities**

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/270002
Waiting https://github.com/NixOS/nixpkgs/pull/269415
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by arthsmn** [comment]
@AndersonTorres the PR is finally merged! You can procede with the update.

**Comment by rhododendrox** [comment]
@AndersonTorres the pr made it's way into master, you can merge this if you want.

**Comment by AndersonTorres** [comment]
No, I need to wait.
For now I'll update.

**Comment by AndersonTorres** [comment]
Shot by splicing again!

**Review by fpletz — APPROVED** [review]

**Comment by fpletz** [comment]
Code looks ok, but darwin fails: https://github.com/NixOS/nixpkgs/pull/270691/checks?check_run_id=20964475886

**Comment by AndersonTorres** [comment]
Fails by not finding libplacebo?
Is this lib available on Darwrn?

**Comment by fpletz** [comment]
libplacebo seems to fail on darwin. Not sure why mpv is searching for this even though vulkan support is only enabled on Linux. I don't have access to a darwin machine to debug this.

**Comment by fpletz** [comment]
Looking at the meson options, this commit could maybe fix the darwin build (cannot push onto your branch): https://github.com/NixOS/nixpkgs/commit/f56f464551f72495d090e0836ec3162275a10776

**Comment by AndersonTorres** [comment]
mental note: fixup authorship of last commit

**Comment by fpletz** [comment]
No worries. Unfortunately, no luck there. :(

**Comment by wegank** [comment]
@ofborg build mpv mpv.passthru.tests

**Comment by davidkna** [comment]
@AndersonTorres [This commit](https://github.com/davidkna/nixpkgs/commit/c19c563b759617e22088b7078a174c6d67877c32) unbreaks the app-bundling for me. Furthermore, the moltenvk PR for `vulkanSupport` has been merged.

**Comment by AndersonTorres** [comment]
Can this patch be upstreamed, at least partially?

**Comment by wegank** [comment]
Looks like `libplacebo` is in fact [unconditional in v0.37.0](https://github.com/mpv-player/mpv/blob/818ce7c51a6b9179307950e919983e0909942098/meson.build#L26)?

**Comment by reckenrode** [comment]
> The python script tries to sign by itself now, but `sigtool` doesn't have `--deep`.

You may want to upstream the changes for this. `--deep` is deprecated in `codesign` as of macOS 13.

**Comment by davidkna** [comment]
I assume that `--deep` would also sign any dylib dependencies if `-s`/`--skip-deps` is not used, making simple removal not a true replacement.

**Comment by reckenrode** [comment]
This is what the man page says on macOS 14. It only signs things if they are in certain places. I don’t know what `--skip-deps` is. It’s not listed in the man page. `-s` is short for `--sign`.

> `--deep` **(DEPRECATED for signing as of macOS 13.0)** When signing a bundle, specifies that nested code content such as helpers, frameworks, and plug-ins,  should be recursively signed in turn.
> 
> **Beware:**
> * All signing options will be applied, in turn, to all nested content. This is almost never what you want.
> * _Nested  code  content_ is  a special term that only applies to macOS style bundles with a Contents folder. Only bare Mach-Os and well structured bundles qualify as _nested code content_. Non-bundle directories in nested code content locations will cause an error when signing.  The `codesign` tool will only discover _nested code content_ in the following directories:
>   * Contents
>   * Contents/Frameworks
>   * Contents/SharedFrameworks
>   * Contents/PlugIns
>   * Contents/Plug-ins
>   * Contents/XPCServices
>   * Contents/Helpers
>   * Contents/MacOS
>   * Contents/Library/Automator
>   * Contents/Library/Spotlight
>   * Contents/Library/LoginItems
> * If any code (Mach-Os, bundles) are located outside the above listed locations they will not be signed by the `--deep` option
> * Using  the `--deep` option on an iOS style bundle without a Contents folder will not cause an error but will only sign the main binary of the bundle.
> 
> When verifying a bundle, this option specifies that any nested code content will be recursively verified as to its full content. By default, verification of nested content is limited to a shallow investigation that may not detect changes to the nested code.
> 
> When displaying a signature, this option specifies that a list of directly nested code should be written to the display output. This lists  only  code directly nested within the subject; anything nested indirectly will require recursive application of the `codesign` command.

**Comment by reckenrode** [comment]
Alternatively, does using `rcodesign` instead of `sigtool` work? It’s not CLI compatible with `codesign`, but if the codesigning command is already being patched, it may do everything needed without having to implement a workaround.

**Comment by reckenrode** [comment]
The following patch to @davidkna’s https://github.com/davidkna/nixpkgs/commit/890f0673723f9c53156c7600d10d4fd3309618cb allows it to use `rcodesign sign` instead. According to the documentation, `sign` will sign nested bundles by default, which should address the concern regarding dropping `--deep`.

I also enabled Vulkan support after rebasing on master, though I don’t know how to tell if it’s working.

```patch
diff --git a/pkgs/applications/video/mpv/darwin-sigtool-no-deep.patch b/pkgs/applications/video/mpv/darwin-sigtool-no-deep.patch
index 35ca32653a36..74ab97db4e54 100644
--- a/pkgs/applications/video/mpv/darwin-sigtool-no-deep.patch
+++ b/pkgs/applications/video/mpv/darwin-sigtool-no-deep.patch
@@ -7,7 +7,7 @@ index 98699e478b..d02ecf610e 100755
  
  def sign_bundle(binary_name):
 -    sh('codesign --force --deep -s - ' + bundle_path(binary_name))
-+    sh('codesign --force -s - ' + bundle_path(binary_name))
++    sh('rcodesign sign ' + bundle_path(binary_name))
  
  def bundle_version():
      if os.path.exists('VERSION'):
diff --git a/pkgs/applications/video/mpv/default.nix b/pkgs/applications/video/mpv/default.nix
index 889def880752..dbccec06f8bc 100644
--- a/pkgs/applications/video/mpv/default.nix
+++ b/pkgs/applications/video/mpv/default.nix
@@ -20,7 +20,7 @@
 , libuchardet
 , libiconv
 , xcbuild
-, sigtool
+, rcodesign
 
 , waylandSupport ? stdenv.isLinux
   , wayland
@@ -156,7 +156,7 @@ in stdenv'.mkDerivation (finalAttrs: {
     ninja
     pkg-config
   ]
-  ++ lib.optionals stdenv.isDarwin [ xcbuild.xcrun sigtool ]
+  ++ lib.optionals stdenv.isDarwin [ xcbuild.xcrun rcodesign ]
   ++ lib.optionals swiftSupport [ swift ]
   ++ lib.optionals waylandSupport [ wayland-scanner ];
 
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 783668e0f23d..4139e71ed69d 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -33880,7 +33880,6 @@ with pkgs;
   mpv-unwrapped = darwin.apple_sdk_11_0.callPackage ../applications/video/mpv {
     stdenv = if stdenv.isDarwin then swiftPackages.stdenv else stdenv;
     inherit lua;
-    inherit (darwin) sigtool; # otherwise it breaks splicing...
   };
 
   shaka-packager = callPackage ../applications/video/shaka-packager { };
```

**Comment by davidkna** [comment]
@reckenrode

>  I don’t know what `--skip-deps` is. It’s not listed in the man page. `-s` is short for `--sign`.

`--skip-deps` is a flag for `osxbundle.py` that the build-script uses, which prevents `osxbundle.py` from copying dynamic library dependencies into the bundle. Given the other description above, signing those files in addition to the main binary might suffice for replacing `--deep`.

> I also enabled Vulkan support after rebasing on master, though I don’t know how to tell if it’s working. 

I successfully tested vulkan support on macOS with `--gpu-api=vulkan --vo=gpu-next`.

**Comment by AndersonTorres** [comment]
Still not working for Darwin...

**Comment by reckenrode** [comment]
It’s failing because libplacebo is conditioned on Vulkan support in the derivation, but the build requires it regardless. It needs to be added unconditionally to mpv’s build inputs. (Whether to enable Vulkan by default on Darwin is a separate question. It appears to work, but I don’t know what upstream expects or requires from it on Darwin.)

**Comment by Tungsten842** [comment]
> It’s failing because libplacebo is conditioned on Vulkan support in the derivation, but the build requires it regardless. It needs to be added unconditionally to mpv’s build inputs. (Whether to enable Vulkan by default on Darwin is a separate question. It appears to work, but I don’t know what upstream expects or requires from it on Darwin.)

Libplacebo supports vulkan on darwin through: [MoltenVK](https://github.com/KhronosGroup/MoltenVK) 
https://code.videolan.org/videolan/libplacebo#hardware-requirements

**Comment by geri1701** [comment]
Result of `nixpkgs-review pr 270691` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>feishin</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>mov-cli</li>
    <li>mov-cli.dist</li>
    <li>sublime-music</li>
    <li>sublime-music.dist</li>
  </ul>
</details>
<details>
  <summary>76 packages built:</summary>
  <ul>
    <li>adl</li>
    <li>ani-cli</li>
    <li>anime-downloader</li>
    <li>anime-downloader.dist</li>
    <li>anki</li>
    <li>anki.dist</li>
    <li>anki.doc</li>
    <li>anki.man</li>
    <li>celluloid</li>
    <li>cplay-ng</li>
    <li>cplay-ng.dist</li>
    <li>curseradio</li>
    <li>curseradio.dist</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>dmlive</li>
    <li>dra-cla</li>
    <li>ff2mpv</li>
    <li>ff2mpv-go</li>
    <li>gonic</li>
    <li>gtk-pipe-viewer</li>
    <li>gtk-pipe-viewer.devdoc</li>
    <li>haruna</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>hypnotix</li>
    <li>invidtui</li>
    <li>jellyfin-media-player</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>jftui</li>
    <li>klipperscreen</li>
    <li>libsForQt5.plasmatube (plasma5Packages.plasmatube)</li>
    <li>libsForQt5.tokodon (plasma5Packages.tokodon)</li>
    <li>linux-wallpaperengine</li>
    <li>memento</li>
    <li>minitube</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mpc-qt</li>
    <li>mpv</li>
    <li>mpv-unwrapped</li>
    <li>mpv-unwrapped.doc</li>
    <li>mpv-unwrapped.man</li>
    <li>mpvScripts.acompressor</li>
    <li>mpvScripts.autocrop</li>
    <li>mpvScripts.autodeint</li>
    <li>mpvScripts.autoload</li>
    <li>mpvScripts.inhibit-gnome</li>
    <li>mpvScripts.mpris</li>
    <li>mpvpaper</li>
    <li>photoqt</li>
    <li>pipe-viewer</li>
    <li>pipe-viewer.devdoc</li>
    <li>plex-media-player</li>
    <li>plex-mpv-shim</li>
    <li>plex-mpv-shim.dist</li>
    <li>python311Packages.mpv</li>
    <li>python311Packages.mpv.dist</li>
    <li>python312Packages.mpv</li>
    <li>python312Packages.mpv.dist</li>
    <li>qimgv</li>
    <li>radioboat</li>
    <li>somafm-cli</li>
    <li>spotube</li>
    <li>stremio</li>
    <li>subtitleedit</li>
    <li>supersonic</li>
    <li>supersonic-wayland</li>
    <li>svp</li>
    <li>tomato-c</li>
    <li>wtwitch</li>
    <li>youtube-tui</li>
    <li>ytfzf</li>
    <li>ytui-music</li>
  </ul>
</details>

**Comment by reckenrode** [comment]
> > It’s failing because libplacebo is conditioned on Vulkan support in the derivation, but the build requires it regardless. It needs to be added unconditionally to mpv’s build inputs. (Whether to enable Vulkan by default on Darwin is a separate question. It appears to work, but I don’t know what upstream expects or requires from it on Darwin.)
> 
> Libplacebo supports vulkan on darwin through: [MoltenVK](https://github.com/KhronosGroup/MoltenVK) https://code.videolan.org/videolan/libplacebo#hardware-requirements

I wasn’t trying to suggest it didn’t or shouldn’t, and I apologize if I gave that impression. I wanted to address the failing build without getting sidetracked discussing what the default for `vulkanSupport` should be in the mpv derivation.

It seems like it probably should be the default, but I’d prefer to defer to the others here who have been giving good feedback and providing fixes for the issues. My only contribution really was the switch to `rcodesign`.

**Comment by AndersonTorres** [comment]
The things are getting worse. Now two packages don't build.

**Comment by wegank** [comment]
@ofborg build sublime-music

**Comment by wegank** [comment]
> The things are getting worse. Now two packages don't build.

`mov-cli` doesn't build on master either.

**Comment by AndersonTorres** [comment]
> > The things are getting worse. Now two packages don't build.
> 
> `mov-cli` doesn't build on master either.

yes, I know. https://github.com/NixOS/nixpkgs/pull/284587 

Now the point is about `sublime-music`.

**Comment by wegank** [comment]
> Now the point is about `sublime-music`.

Looks good according to ofborg, probably some flaky tests failed/timed out?

**Review by wegank — APPROVED** [review]

**Review by geri1701 — APPROVED** [review]

File: pkgs/applications/video/mpv/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/video/mpv/default.nix:105]: Why remove the dev output?

Comment (AndersonTorres) [reply pkgs/applications/video/mpv/default.nix:105]: How?

Comment (SuperSandro2000) [reply pkgs/applications/video/mpv/default.nix:105]: I did nix build --keep-failed and noticed that they code the dev output into the shared objects.

**Review by fpletz — APPROVED** [review]

**Pull request merged**
